cloudformation/ad/ad-integration.yaml

```diff
@@ -373,43 +373,44 @@ Resources:
           !Sub
             - |
               #!/bin/bash -e
+              set -o pipefail
               exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
               yum update -y aws-cfn-bootstrap
-              /opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource AdDomainAdminNode --configsets setup --region ${AWS::Region}
-              echo "Domain Name: " ${DirectoryDomain}
-              echo "Domain Certificate Secret: " ${DomainCertificateSecretArn}
-              echo "Domain Private Key Secret: " ${DomainPrivateKeySecretArn}
+              /opt/aws/bin/cfn-init -v --stack "${AWS::StackName}" --resource AdDomainAdminNode --configsets setup --region "${AWS::Region}"
+              echo "Domain Name: ${DirectoryDomain}"
+              echo "Domain Certificate Secret: ${DomainCertificateSecretArn}"
+              echo "Domain Private Key Secret: ${DomainPrivateKeySecretArn}"
               cat << EOF > /etc/resolv.conf
               search           ${DirectoryDomain}
               nameserver       ${DnsIp1}
               nameserver       ${DnsIp2}
               EOF
               sed -i 's/PEERDNS=.*/PEERDNS=no/' /etc/sysconfig/network-scripts/ifcfg-eth0
               chattr +i /etc/resolv.conf
-              ADMIN_PW=${AdminPassword}
-              echo $ADMIN_PW | sudo realm join -U Admin ${DirectoryDomain}
+              ADMIN_PW="${AdminPassword}"
+              echo "$ADMIN_PW" | sudo realm join -U Admin "${DirectoryDomain}"
```

Comment:
nit: might wrap these in `{` for consistency with other variables.

Reply:
Done

Reply:
I think `ADMIN_PW` is not wrapped with `{}` because `{}`-wrapped variables are those injected by `Fn::Sub`, whereas the unwrapped ones are those declared within the script.

Reply:
Ah -- I believe I missed that context :+1: